api/api.go

```diff
@@ -267,9 +268,78 @@ func (apiResource ApiWrapper) SearchOrganizations(ctx echo.Context, params Searc
 func (apiResource ApiWrapper) MTLSCAs(ctx echo.Context) error {
 	var err error
 
+	CAs := apiResource.R.VendorCAs()
+
+	acceptHeader := ctx.Request().Header.Get("Accept")
+	if "application/json" == acceptHeader {
+		result := cAsAsCAListWithChain(CAs)
+		ctx.JSON(http.StatusOK, &result)
+	} else { // otherwise application/x-pem-file
+		ctx.Response().Header().Set("Content-Type", "application/x-pem-file")
+		ctx.String(http.StatusOK, cAsAsSinglePEM(CAs))
+	}
+
 	return err
 }
 
+// cAsAsSinglePEM sorts the given string by depth
+func cAsAsSinglePEM(CAs [][]string) string {
+	var depthSet []map[string]bool
+	size := 0 // memory optimization
+	for _, ch := range CAs {
+		for j, c := range ch {
+			if len(depthSet) <= j {
+				depthSet = append(depthSet, map[string]bool{})
+			}
+			currentDepthSet := depthSet[j]
+			if !currentDepthSet[c] {
+				currentDepthSet[c] = true
+				size += len(c) + 1 // memory optimization (newline)
+			}
+		}
+	}
+	var result strings.Builder
+	result.Grow(size) // memory optimization
+	// now iterate per depth
+	for _, s := range depthSet {
```

Comment:
I believe we're returning duplicate entries now? (since we probably 1 root + 1 intermediate and many/a few vendor CA entries). Since a PEM sort of unstructured (or at least, it doesn't have a notion of 'multiple chains' as far as I know. Just that the first certificate is the leaf, then intermediate, etc etc) I don't think we should have duplicate entries (think I've never seen PEM bundles with duplicate entries).

Reply:
nope, they are sets or the Go equivalent: `map[string]bool` and I iterate over the keys